pipenv/environment.py

```diff
@@ -712,6 +713,23 @@ def is_installed(self, pkgname):
 
         return any(d for d in self.get_distributions() if d.project_name == pkgname)
 
+    def is_satisfied(self, req):
+        match = next(
+            iter(
+                d for d in self.get_distributions()
+                if canonicalize_name(d.project_name) == req.normalized_name
+            ), None
+        )
+        if match is not None:
+            if req.editable and self.find_egg(match):
+                return req.line_instance.path == match.location
```

Comment:
Use `os.path.samefile` here instead?

Reply:
`samefile` is unix only afaik

Reply:
More precisely, it’s Unix-only on Python 2. I would use it if possible (and fallback to `==` otherwise)

Reply:
It looks like `samefile` just compares `os.stat` outputs which should still be valid on python 2, right? Is that still ok on windows?

```python
def samefile(f1, f2):
    """Test whether two pathnames reference the same actual file or directory
    This is determined by the device number and i-node number and
    raises an exception if an os.stat() call on either pathname fails.
    """
    s1 = os.stat(f1)
    s2 = os.stat(f2)
    return samestat(s1, s2)


def samestat(s1, s2):
    """Test whether two stat buffers reference the same file"""
    return (s1.st_ino == s2.st_ino and
            s1.st_dev == s2.st_dev)
```

Reply:
We won’t know unless we try it out 😉

Reply:
It works fine on `vistir`'s test suite. `vistir`'s subprocess runner has a threading race condition I am ignoring for now to get this release out, but those are the only failures now.